retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java

```diff
@@ -0,0 +1,207 @@
+        requireNonNull(baseUrl, "baseUrl");
+        String path = URI.create(baseUrl).getPath();
+        if (!path.isEmpty() && !SLASH.equals(path.substring(path.length() - 1))) {
+            throw new IllegalArgumentException("baseUrl must end in /: " + baseUrl);
+        }
+        this.baseUrl = baseUrl;
+        return this;
+    }
+
+    /**
+     * Sets the {@link BiFunction} that is applied to the {@link ClientOptions} of the underlying
+     * {@link HttpClient}.
+     *
+     * @param configurator a {@link Function} whose input is the original {@link ClientOptions} of the client
+     *                     being derived from and whose output is the {@link ClientOptions} of the new derived
+     *                     client
+     */
+    public ArmeriaRetrofitBuilder withClientOptions(
+            BiFunction<String, ? super ClientOptions, ClientOptions> configurator) {
+        requireNonNull(configurator, "configurator");
+        this.configurator = configurator;
+        return this;
+    }
+
+    /**
+     * Adds the specified converter factory for serialization and deserialization of objects.
+     * @see Retrofit.Builder#addCallAdapterFactory(Factory)
+     */
+    public ArmeriaRetrofitBuilder addConverterFactory(Converter.Factory factory) {
+        requireNonNull(factory, "factory");
+        retrofitBuilder.addConverterFactory(factory);
+        return this;
+    }
+
+    /**
+     * Adds the specified call adapter factory for supporting service method return types other than {@link
+     * Call}.
+     * @see Retrofit.Builder#addCallAdapterFactory(Factory)
+     */
+    public ArmeriaRetrofitBuilder addCallAdapterFactory(CallAdapter.Factory factory) {
+        requireNonNull(factory, "factory");
+        retrofitBuilder.addCallAdapterFactory(factory);
+        return this;
+    }
+
+    /**
+     * Sets the {@link Executor} on which {@link Callback} methods are invoked when returning {@link Call} from
+     * your service method.
+     *
+     * <p>Note: {@code executor} is not used for {@link #addCallAdapterFactory custom method return types}.
+     *
+     * @see Retrofit.Builder#callbackExecutor(Executor)
+     */
+    public ArmeriaRetrofitBuilder callbackExecutor(Executor executor) {
+        requireNonNull(executor, "executor");
+        retrofitBuilder.callbackExecutor(executor);
+        return this;
+    }
+
+    /**
+     * When calling {@link Retrofit#create} on the resulting {@link Retrofit} instance, eagerly validate
+     * the configuration of all methods in the supplied interface.
+     *
+     * @see Retrofit.Builder#validateEagerly(boolean)
+     */
+    public ArmeriaRetrofitBuilder validateEagerly(boolean validateEagerly) {
+        retrofitBuilder.validateEagerly(validateEagerly);
+        return this;
+    }
+
+    /**
+     * Creates a new {@link Retrofit} instance using the configured values.
+     */
+    public Retrofit build() {
+        requireNonNull(baseUrl, "baseUrl");
```

Comment:
This should raise an `IllegalStateException`. You could use `Preconditions.checkArgument()`. Alternatively, you could accept the `baseUrl` as a constructor parameter?